types/puppeteer/index.d.ts

```diff
@@ -1279,40 +1361,19 @@ export interface FrameBase extends Evalable {
     ...args: SerializableOrJSHandle[]
   ): Promise<JSHandle>;
 
-  /**
-   * Allows waiting for various conditions.
-   */
-  waitForFunction(
-    fn: EvaluateFn,
-    options?: PageFnOptions,
-    ...args: SerializableOrJSHandle[]
-  ): Promise<JSHandle>;
-
   /**
    * Wait for the page navigation occur.
    * @param options The navigation parameters.
    */
   waitForNavigation(options?: NavigationOptions): Promise<Response>;
-
-  waitForSelector(
-    selector: string,
-    options?: WaitForSelectorOptions,
-  ): Promise<ElementHandle>;
-  waitForSelector(
-      selector: string,
-      options?: WaitForSelectorOptionsHidden,
-  ): Promise<ElementHandle | null>;
-
-  waitForXPath(
-    xpath: string,
-    options?: WaitForSelectorOptions,
-  ): Promise<ElementHandle>;
 }
 
-export interface Frame extends FrameBase {
+
+export interface Frame extends Contentable, Evalable, Evaluateable, ExecutionContextable, FrameBase {
+  // constructor(frameManager: FrameManager, client: CDPSession, parentFrame: Frame | null | undefined, frameId: string);
+  _mainWorld: DOMWorld;
```

Comment:
We shouldn't expose this.

Reply:
oups I forgot to remove this one, (this was the only one)